src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -1520,10 +1546,28 @@ def to_series(self) -> None:
         Index.to_frame : Convert an Index to a DataFrame.
         Series.to_frame : Convert Series to DataFrame.
         """
-        # TODO: SNOW-1458117 implement to_series
+        from snowflake.snowpark.modin.pandas import Series
 
-    @index_not_implemented()
-    def to_frame(self) -> None:
+        # get the index name if the name is not given
+        if name is None:
+            name = self.name
+
+        # convert self to a dataframe and get qc
+        # this will give us a df where the index and data columns both have self
+        new_qc = self.to_frame(name=name)._query_compiler
+
+        # if we are given an index, join this index column into qc
+        if index is not None:
+            new_qc = new_qc.set_index_from_series(Series(index)._query_compiler)
+
+        # create series and set the name
+        ser = Series(query_compiler=new_qc)
+        ser.name = name
+        return ser
+
+    @is_lazy_check
+    # TODO: SNOW-1481037 : Fix typehints
+    def to_frame(self, index: bool = True, name: Hashable | None = None) -> Any:
```

Comment:
you mentioned that you were using Any because it got reverted, right? if you quote your return type like "DataFrame" does it automatically revers?

Reply:
yes it automatically removes quotes and then causes errors